pkg/kubelet/kubelet_pods.go

```diff
@@ -326,7 +327,10 @@ func ensureHostsFile(fileName, hostIP, hostName, hostDomainName string, hostAlia
 		hostsFileContent = managedHostsFileContent(hostIP, hostName, hostDomainName, hostAliases)
 	}
 
-	return ioutil.WriteFile(fileName, hostsFileContent, 0644)
+	oldMask, _ := util.Umask(0)
```

Comment:
It feels a little weird that we are doing nothing when there is an error... could we try and do some intelligent form of logging? Either that we failed to set the Umask, or the existing platform doesn't support the Umask?

Reply:
Also, I think if the platform doesn't support `Umask`, we will still want to perform the check at the end that the file has the correct write permission.